src/Fabulous.XamarinForms/Views/Controls/ImageButton.fs

```diff
@@ -1,39 +1,168 @@
 namespace Fabulous.XamarinForms
 
+open System.IO
+open System.Runtime.CompilerServices
 open Fabulous
 open Xamarin.Forms
+open System
 
 type IImageButton =
     inherit IView
 
 module ImageButton =
     let WidgetKey = Widgets.register<ImageButton> ()
 
-    let Source =
-        Attributes.defineBindable<ImageSource> ImageButton.SourceProperty
-
     let Aspect =
         Attributes.defineBindable<Xamarin.Forms.Aspect> ImageButton.AspectProperty
 
+    let BorderColor =
+        Attributes.defineAppThemeBindable<Color> ImageButton.BorderColorProperty
+
+    let BorderWidth =
+        Attributes.defineBindable<float> ImageButton.BorderWidthProperty
+
+    let CornerRadius =
+        Attributes.defineBindable<float> ImageButton.CornerRadiusProperty
+
+    let IsLoading =
+        Attributes.defineBindable<bool> ImageButton.IsLoadingProperty
+
+    let IsOpaque =
+        Attributes.defineBindable<bool> ImageButton.IsOpaqueProperty
+
+    let IsPressed =
+        Attributes.defineBindable<bool> ImageButton.IsPressedProperty
+
+    let Padding =
+        Attributes.defineBindable<Thickness> ImageButton.PaddingProperty
+
+    let Source =
+        Attributes.defineAppThemeBindable<ImageSource> ImageButton.SourceProperty
+
     let Clicked =
         Attributes.defineEventNoArg "ImageButton_Clicked" (fun target -> (target :?> ImageButton).Clicked)
 
+    let Pressed =
+        Attributes.defineEventNoArg "ImageButton_Pressed" (fun target -> (target :?> ImageButton).Pressed)
+
+    let Released =
+        Attributes.defineEventNoArg "ImageButton_Released" (fun target -> (target :?> ImageButton).Released)
+
 [<AutoOpen>]
 module ImageButtonBuilders =
     type Fabulous.XamarinForms.View with
-        static member inline ImageButton<'msg>(source: ImageSource, onClicked: 'msg, aspect: Xamarin.Forms.Aspect) =
-            WidgetBuilder<'msg, IImageButton>(
-                ImageButton.WidgetKey,
-                ImageButton.Source.WithValue(source),
-                ImageButton.Clicked.WithValue(onClicked),
-                ImageButton.Aspect.WithValue(aspect)
-            )
-
-        static member inline ImageButton<'msg>(path: string, onClicked, aspect) =
-            View.ImageButton<'msg>(ImageSource.FromFile(path), onClicked, aspect)
-
-        static member inline ImageButton<'msg>(uri: System.Uri, onClicked, aspect) =
-            View.ImageButton<'msg>(ImageSource.FromUri(uri), onClicked, aspect)
-
-        static member inline ImageButton<'msg>(stream: System.IO.Stream, onClicked, aspect) =
-            View.ImageButton<'msg>(ImageSource.FromStream(fun () -> stream), onClicked, aspect)
+        static member inline ImageButton<'msg>
+            (
+                light: ImageSource,
+                onClicked: 'msg,
+                ?dark: ImageSource,
+                ?aspect: Aspect
```

Comment:
I would make `aspect` mandatory and the 1st parameter to be consistent with other controls

Reply:
Tbh I think having the aspect as the first parameters felt confusing to me 😅. I was thinking if we could make it optional in both the Image and the imagebutton . So the Image will be (light , ?dark, ?aspect)

Reply:
Updated Following your suggestion ;)

Reply:
Thanks! The more we talk about it, the more I think my idea of making aspect mandatory might not be the best.
Maybe we could change that in another PR if that's something you would prefer.